server-api/src/main/java/org/sonarsource/sonarlint/core/serverapi/issue/IssueApi.java

```diff
@@ -264,8 +264,14 @@ public CompletableFuture<Issue> searchByKey(String issueKey) {
       });
   }
 
-  public Optional<ServerIssueDetails> fetchServerIssue(String issueKey) {
-    String searchUrl = "/api/issues/search.protobuf?issues=" + urlEncode(issueKey) + "&ps=1&p=1";
+  public Optional<ServerIssueDetails> fetchServerIssue(String issueKey, String projectKey, String branch, @Nullable String pullRequest) {
+    String searchUrl = "/api/issues/search.protobuf?issues=" + urlEncode(issueKey) + "&componentKeys=" + projectKey + "&ps=1&p=1";
+    if (pullRequest != null && !pullRequest.isEmpty()) {
+      searchUrl = searchUrl.concat("&pullRequest=").concat(pullRequest);
```

Comment:
Probably safer to url encode the `pullRequest` value, not sure what kind of values it can hold